### ISSUE TITLE
Function exists checks are not passing the checker

### ISSUE THREAD

**Issue by hkdobrev** [maglnet/ComposerRequireChecker#65]
I have code similar to:

```php
if (function_exists('xdebug_get_function_stack')) {
	$trace = array_slice(array_reverse(xdebug_get_function_stack()), 4);
}
```

So running this with XDebug extension would get the trace differently, but it could work without XDebug. There could be other conditional uses of a function or a class, especially for backwards-compatibility.

However, the checker complains about these function uses as the parses doesn't detect the `function_exists` condition.

**Comment by hkdobrev** [comment]
`class_exists` checks are also quite common and would probably also suffer from the same issue.

**Comment by Ocramius** [comment]
`function_exists()` is a runtime expression with another runtime expression
as parameter. This can only be used if:

 * The parameter is a constant expression
 * The `\function_exists` symbol is referenced absolutely or via `use
function`.

In either way, that is not something this library should go in, as the
complexity explodes massively, and every conditional leads to two
separately checked scopes.

**Comment by VincentLanglet** [comment]
> `class_exists` checks are also quite common and would probably also suffer from the same issue.

I just encounter the issue with code like
```
        if (class_exists(CacheTokenParser::class)) {
            $this->addTokenParser(new CacheTokenParser());
        }
```
It's the only use of CacheTokenParser and it's reported as an error by composer-require-checker.

But not sure it can be solved.

**Comment by Ocramius** [comment]
@VincentLanglet that's something you'd exclude manually via config.
